### ISSUE TITLE
[Bug] basic typescript integration fails with "no exported member" errors 

### ISSUE THREAD

**Issue by davidbielik** [berstend/puppeteer-extra#428]
**Describe the bug**
[Minimum Reproducible Code](https://github.com/berstend/puppeteer-extra/files/5975973/Archive.zip)

1. unzip this file
2. run `npm install`
3. run `tsc`

Expected: typescript compiles
Actual: lots of "namespace has no exported member" errors

This is just from importing the library. When I have actual puppeteer code i get many more errors for lots of other un-exported modules.

```
node_modules/puppeteer-extra-plugin/dist/index.d.ts:315:39 - error TS2694: Namespace '"/puppeteer-extra-bug/node_modules/puppeteer/lib/types"' has no exported member 'Worker'.

315     onWorkerCreated(worker: Puppeteer.Worker): Promise<void>;
                                          ~~~~~~

node_modules/puppeteer-extra/dist/puppeteer.d.ts:6:10 - error TS2305: Module '"../../puppeteer/lib/types"' has no exported member 'FetcherOptions'.

6 export { FetcherOptions } from 'puppeteer';
           ~~~~~~~~~~~~~~

[10:54:40 AM] Found 2 errors. Watching for file changes.
```

**Code Snippet**

```javascript
import puppeteer from "puppeteer-extra";
import StealthPlugin from "puppeteer-extra-plugin-stealth";
puppeteer.use(StealthPlugin());
```

**Versions**
```
System:
    OS: macOS 10.15.5
    CPU: (8) x64 Intel(R) Core(TM) i7-1068NG7 CPU @ 2.30GHz
    Memory: 139.73 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 10.23.1 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.10 - /usr/local/bin/npm
  npmPackages:
    puppeteer: ^7.1.0 => 7.1.0 
    puppeteer-extra: ^3.1.16 => 3.1.16 
    puppeteer-extra-plugin-stealth: ^2.7.4 => 2.7.4 
```

**Comment by berstend** [comment]
@davidbielik puppeteer broken typings, just look at this mess:
https://github.com/puppeteer/puppeteer/releases

Use `puppeteer@5` for now.

**Comment by berstend** [comment]
I've added a note in the main readme (in the hopes people will see it):

![image](https://user-images.githubusercontent.com/1368633/107919713-11c1d980-6f6c-11eb-9f75-55d94bfd5cd7.png)

**Comment by berstend** [comment]
Just to track some issues regarding the upstream puppeteer typing issues:
https://github.com/puppeteer/puppeteer/issues/6899
https://github.com/puppeteer/puppeteer/issues/6885

**Comment by 1nVitr0** [comment]
Yeah it's a real tragedy. Of  100 types the `@types/puppeteer` package originally exported, 51 are no longer available with an additional 23 that no longer match the original types for various reasons. And that's just the top level type exports.

To burn through 3 major versions in a matter of a month is not something you see everyday...

**Comment by berstend** [comment]
> Yeah it's a real tragedy. Of 100 types the `@types/puppeteer` package originally exported, 51 are no longer available with an additional 23 that no longer match the original types for various reasons. And that's just the top level type exports.
> 
> To burn through 3 major versions in a matter of a month is not something you see everyday...

100% agree. It's really painful to the point where I halted the #303 release until they sorted this out.

**Comment by anphetamina** [comment]
Using puppeteer and types-puppeteer 5.4
![image](https://user-images.githubusercontent.com/23299902/112517595-c5eb2700-8d98-11eb-89fe-40b9089b7a7f.png)

**Comment by PairedPrototype** [comment]
> Solution: Use `puppeteer@5` for now.

Was this ever confirmed to work? It's still completely broken 😢 I tried v5.0.0 and v5.5.0

**Comment by 1nVitr0** [comment]
> Was this ever confirmed to work? It's still completely broken  I tried v5.0.0 and v5.5.0

I've been working with @5 before puppeteer came and broke their typings and it still works fine. Did you remove the @6+ version of puppeteer from the `node_modules` folder and include `@types/puppeteer` as a dev dependency?

**Comment by PairedPrototype** [comment]
> I've been working with `@5` before puppeteer came and broke their typings and it still works fine. Did you remove the `@6+` version of puppeteer from the `node_modules` folder and include `@types/puppeteer` as a dev dependency?

I realised what the issue was, this package on at least that version doesn't appear to have complete coverage of passing through the types provided by `@types/puppeteer` so, at face value, I just saw redline in VSCode and instantly came to the conclusion it was broken. 

I was wrong, it was just some types that didn't work through the `puppeteer-extra` package, I just instead cast them as the types I knew they were and things seemed to work again 🙂 I'm not sure if there's a more well-established solution for this, but I haven't bothered to check either, it was just for a quick and dirty script, for now at least 😁

**Comment by miramardesign** [comment]
@PairedPrototype  can you provide an example? of your fix?  Should i import puppeteer extra and cast somehow?
im getting something like:  { bindAddress: string; executablePath: string; args: string[]; }' is not assignable to parameter of type 'LaunchOptions

**Comment by PairedPrototype** [comment]
@miramardesign Hey, sorry for the delayed reply.

I've created a working minimal-ish [example repo](https://github.com/PairedPrototype/puppeteer-extra-v3.1.18-with-puppeteer-v5.4.1-example) showing how I used the package to achieve my goal. I hope it helps 🙂

**Comment by mmelnic** [comment]
still problems with typescript and puppeteer 10.1.0?

**Comment by berstend** [comment]
This should now be fixed through #555:
```
Successfully published:
 - puppeteer-extra-plugin-adblocker@2.12.0
 - puppeteer-extra-plugin-anonymize-ua@2.3.0
 - puppeteer-extra-plugin-block-resources@2.3.0
 - puppeteer-extra-plugin-click-and-wait@2.3.0
 - puppeteer-extra-plugin-devtools@2.3.0
 - puppeteer-extra-plugin-flash@2.3.0
 - puppeteer-extra-plugin-font-size@2.3.0
 - puppeteer-extra-plugin-recaptcha@3.5.0
 - puppeteer-extra-plugin-repl@2.3.0
 - puppeteer-extra-plugin-stealth@2.8.0
 - puppeteer-extra-plugin-user-data-dir@2.3.0
 - puppeteer-extra-plugin-user-preferences@2.3.0
 - puppeteer-extra-plugin@3.2.0
 - puppeteer-extra@3.2.0
```
